tests/Acoustics.Test/AudioAnalysisTools/HarmonicAnalysis/HarmonicAlgorithmTests.cs

```diff
@@ -0,0 +1,150 @@
+// <copyright file="HarmonicAnalysisTests.cs" company="QutEcoacoustics">
+// All code in this file and all associated files are the copyright and property of the QUT Ecoacoustics Research Group (formerly MQUTeR, and formerly QUT Bioacoustics Research Group).
+// </copyright>
+
+namespace Acoustics.Test.AudioAnalysisTools.HarmonicAnalysis
+{
+    using System;
+    using System.IO;
+    using System.Linq;
+    using Acoustics.Test.TestHelpers;
+    using global::AnalysisPrograms.Recognizers.Base;
+    using global::AudioAnalysisTools;
+    using global::AudioAnalysisTools.DSP;
+    using global::AudioAnalysisTools.Events;
+    using global::AudioAnalysisTools.StandardSpectrograms;
+    using global::AudioAnalysisTools.WavTools;
+    using global::TowseyLibrary;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+    [TestClass]
+    public class HarmonicAlgorithmTests : OutputDirectoryTest
+    {
+        private static readonly FileInfo TestAsset = PathHelper.ResolveAsset("harmonic.wav");
+        private readonly SpectrogramStandard spectrogram;
+
+        private readonly double decibelThreshold = 6.0;
+        private readonly NoiseReductionType noiseReductionType = NoiseReductionType.Standard;
+
+        // can also try these parameters when testing.
+        //private readonly double decibelThreshold = -80;
+        //private readonly NoiseReductionType noiseReductionType = NoiseReductionType.None;
+
+        public HarmonicAlgorithmTests()
+        {
+            var recording = new AudioRecording(TestAsset);
+            this.spectrogram = new SpectrogramStandard(
+                new SonogramConfig
+                {
+                    WindowSize = 512,
+                    WindowStep = 512,
+                    WindowOverlap = 0,
+                    NoiseReductionType = this.noiseReductionType,
+                    NoiseReductionParameter = 0.0,
+                    Duration = recording.Duration,
+                    SampleRate = recording.SampleRate,
+                },
+                recording.WavReader);
+        }
+
+        [TestMethod]
+        public void TestHarmonicsAlgorithmOn440HertzHarmonic()
+        {
+            var parameters = new HarmonicParameters
+            {
+                MinHertz = 400,
+                MaxHertz = 6000,
```

Comment:
The harmonic detection should work without the top and bottom frequency bounds being tight. In fact it is a desirable feature that it should perform well with loose frequency bounds. The TestHarmonicsAlgorithm() in GenericRecognizerTests works well with loose very loose bounds. But this is an artificial example. Your test recording is more difficult to deal with because there appears to be a lot of "structured noise" in the background.

Reply:
I tested this code with the original values - it almost works.